journal.py

```diff
@@ -11,107 +10,136 @@
-
-env = Environment(loader=PackageLoader('journal', 'templates'))
-
-here = os.path.dirname(os.path.abspath(__file__))
-
-DB_SCHEMA = """
-CREATE TABLE IF NOT EXISTS entries (
-    id serial PRIMARY KEY,
-    title VARCHAR (127) NOT NULL,
-    text TEXT NOT NULL,
-    created TIMESTAMP NOT NULL
+import sqlalchemy as sa
+from zope.sqlalchemy import ZopeTransactionExtension
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import (
+    scoped_session,
+    sessionmaker,
 )
-"""
 
-INSERT_ENTRY = """
-INSERT INTO entries (title, text, created) VALUES (%s, %s, %s)
-"""
 
-DB_ENTRIES_LIST = """
-SELECT id, title, text, created FROM entries ORDER BY created DESC
-"""
+env = Environment(loader=PackageLoader('journal', 'templates'))
 
-SELECT_ID = """
-SELECT * FROM entries WHERE id=%s
-"""
+here = os.path.dirname(os.path.abspath(__file__))
 
-READ_ENTRY = """
-SELECT id, title, text, created FROM entries WHERE id = %s
-"""
+# DB_SCHEMA = """
+# CREATE TABLE IF NOT EXISTS entries (
+#     id serial PRIMARY KEY,
+#     title VARCHAR (127) NOT NULL,
+#     text TEXT NOT NULL,
+#     created TIMESTAMP NOT NULL
+# )
+# """
+
+# INSERT_ENTRY = """
+# INSERT INTO entries (title, text, created) VALUES (%s, %s, %s)
+# """
+
+# DB_ENTRIES_LIST = """
+# SELECT id, title, text, created FROM entries ORDER BY created DESC
+# """
+
+# SELECT_ID = """
+# SELECT * FROM entries WHERE id=%s
+# """
+
+# READ_ENTRY = """
+# SELECT id, title, text, created FROM entries WHERE id = %s
+# """
+
+# UPDATE_ENTRY = """
+# UPDATE entries SET (title, text) = (%s, %s) WHERE id=%s
+# """
+
+# NEW_ENTRY = """
+# SELECT * FROM entries ORDER BY created DESC LIMIT 1
+# """
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+Base = declarative_base()
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
 
-UPDATE_ENTRY = """
-UPDATE entries SET (title, text) = (%s, %s) WHERE id=%s
-"""
+    def __repr__(self):
+        return u"{}: {}".format(self.__class__.__name__, self.title)
+
+    @classmethod
+    def all(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def by_id(cls, id):
+        return DBSession.query(cls).filter(cls.id == id).one()
+
+    @classmethod
+    def from_request(cls, request):
+        title = request.params.get('title', None)
+        text = request.params.get('text', None)
+        created = datetime.datetime.utcnow()
+        new_entry = cls(title=title, text=text, created=created)
+        DBSession.add(new_entry)
+
+    @classmethod
+    def desc_new(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).first()
```

Comment:
`newest` might be a more appropriate name for this method.  Try to name things descriptively.